shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/networking/OpenshiftNetworkingService.java

```diff
@@ -52,7 +52,7 @@ public NetworkResource fetchOrCreateBrokerNetworkIngress(BridgeIngress bridgeIng
                 .withName(bridgeIngress.getMetadata().getName())
                 .get();
 
-        if (existing == null || !expected.getSpec().getTo().getName().equals(existing.getSpec().getTo().getName())) {
+        if (existing == null || !existing.getSpec().equals(expected.getSpec())) {
```

Comment:
OpenShift adjusts Route spec definition with default parameters if not defied (i.e. host, to.weight, wildcardPolicy).
Such change will IMHO cause infinite reconciliation.
I think we need to check specific parameters.

Reply:
@sutaakar Is there a definitive set of fields we should check?

We used to compare `To` as a whole (and not properties excluding `to.weight` so we have a problem already.

I really could not find anything to suggest what specific properties we should check.

Even `To` (as defined in our template) is, as you write, not entirely comparable.

Reply:
In previous version I see that we compare just `to.name`.